prql-compiler/tests/integration-rdbms/main.rs

```diff
@@ -0,0 +1,166 @@
+#![cfg(not(any(target_family = "windows", target_family = "wasm")))]
+
+mod connection;
+
+#[cfg(test)]
+mod tests {
+    use std::env;
+
+    use insta::assert_display_snapshot;
+    use postgres::NoTls;
+    use tiberius::{AuthMethod, Client, Config};
+    use tokio::net::TcpStream;
+    use tokio::runtime::Runtime;
+    use tokio_util::compat::{Compat, TokioAsyncWriteCompatExt};
+
+    use prql_compiler::sql::Dialect;
+    use prql_compiler::Options;
+    use prql_compiler::Target::Sql;
+
+    use crate::connection::*;
+
+    #[test]
+    fn test_rdbms() {
+        for port in [5432u16, 3306, 1433 /*, 50000*/] {
+            // test is skipped locally when DB is not listening
+            // in CI it fails
+            if !is_port_open(port) {
+                match env::var("CI") {
+                    Ok(v) if &v == "true" => {
```

Comment:
I don't think we should delay merging to discuss this — it's very easy to change later — but I'm in a couple of minds about this. On the one hand, it's important that `cargo test` passes without external dependencies. On the other, it might be quite confusing if CI is failing from running `cargo test` while locally it passes.

I thought your previous approach re ignoring and then running it explicitly was interesting. Both have their tradeoffs.

Another approach would be to include the out-of-process connections as part of a non-default feature, such that running `cargo test --all-features` ran it, but running `cargo test` didn't. That would probably be my vote, but I haven't thought about it enough. (and again — totally fine to resolve later)

CC @aljazerzen for any views.

Reply:
> On the other, it might be quite confusing if CI is failing from running cargo test while locally it passes.

Yes, but it is the same when the CI runs includes ignored tests

Reply:
Yes totally. It is a bit more legible if it's being passed as a command than implicit in the rust code. But regardless, I think doing it as a feature would dominate both options, and let us split out the external vs. in-process tests

(but as above, fine to skip over for this merge)